src/text/mod.rs

```diff
@@ -1,14 +1,173 @@
-//! Text drawable.
-//!
-//! TODO: - Add an overview of the e-g text rendering, e.g. what a `Text` object is used for and
-//!         how it can be used with different text styles/renderers.
-//!       - Describe the difference between `TextStyle` and `MonoTextStyle`
-//!       - How is `Text::position` used by text renders dependent on `TextStyle::baseline`
-//!       - Add examples:
-//!         - Basic text without `TextStyle`
-//!         - Advanced text with `TextStyle`
-//!         - Draw text with multiple styles by using the result of `Text.draw(...)`
-//!       - Link to `renderer` module docs for users who want to implement custom renderers.
+//! Text drawing.
+//!
+//! The [`Text`] drawable can be used to draw text on a draw target. To construct a [`Text`] object
+//! at least a text string, position and character style are required. For advanced formatting
+//! options an additional [`TextStyle`] object might be required.
+//!
+//! Text rendering in embedded-graphics is designed to be extendable by text renderers for different
+//! font formats. To use a text renderer in an embedded-graphics project each renderer provides a
+//! character style object. This object is used to set the appearance of characters, like the text
+//! color or the used font. The available settings vary between different text renderer and are
+//! documented in the text renderer documentation.
+//!
+//! See the [`renderer` module] docs for more information about implementing custom text renderers.
+//!
+//! Embedded-graphics includes a text renderer for monospaced fonts in the [`mono_font`] module.
+//! Most examples will use this renderer and the associated [`MonoTextStyle`] character style.
+//! But they should be easily adaptable to any external renderer listed in the
+//! [external crates list].
+//!
+//! # Text style
+//!
+//! In addition to styling the individual characters the [`Text`] drawable also contains a
+//! [`TextStyle`] setting. The text style is used to set the alignment and line spacing of text
+//! objects.
+//!
+//! The [`alignment`] setting sets the horizontal alignment of the text. With the default value
+//! `Left` the text will be rendered to the right of the given text position. Analogously `Right`
+//! aligned text will be rendered to the left of the given position. `Center`ed text will extend
+//! equally to the left and right of the text position. If the text contains multiple lines each
+//! line will be aligned independently of the other lines.
```

Comment:
I'm not sure what 

> If the text contains multiple lines each line will be aligned independently of the other lines.

means. It seems to imply I'd have to set the alignment for each line of text. I don't think that's correct, but that's how I understood this sentence.

Reply:
It was supposed to mean that each line would be centered independently and not handled as a centered block of left aligned lines:
![grafik](https://user-images.githubusercontent.com/226123/114740410-1cb9a000-9d4a-11eb-82a5-b2141e6deba3.png)
The green line is the X position passed to `Text::new`.

But I guess that everyone would expect centering to work this way, so I've just removed the confusing sentence.

Reply:
Ah I see now, yeah it's probably best to let assumptions do the work here.